src/Domain/Repository/ChannelPricingLogEntryRepositoryInterface.php

```diff
@@ -22,4 +23,16 @@ interface ChannelPricingLogEntryRepositoryInterface extends RepositoryInterface
      * @return array|ChannelPricingLogEntryInterface[]
      */
     public function findOlderThan(\DateTimeInterface $date, ?int $limit = null): array;
+
+    public function findTheMostRecentOne(ChannelPricingInterface $channelPricing): ?ChannelPricingLogEntryInterface;
+
+    public function findOneWithTheLowestPriceInPeriod(
+        ChannelPricingLogEntryInterface $channelPricingLogEntry,
+        int $days,
+    ): ?ChannelPricingLogEntryInterface;
+
+    public function findTheMostRecentOneBeyondPeriod(
```

Comment:
```suggestion
    public function findLatestOneBeyondPeriod(
```
but I guess that this method could be no longer needed